proposals/csharp-12.0/collection-expressions.md

```diff
@@ -219,7 +213,34 @@ ImmutableArray<int> ia =
 ## Construction
 [construction]: #construction
 
-*Give specific ordering for determining how to construct the constructible collection types.*
+A collection expression has a *known length* if the compile-time type of each *spread element* in the collection expression is [*countable*](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-8.0/ranges.md#adding-index-and-range-support-to-existing-library-types).
+The *countable* properties, `Length` and `Count`, are assumed to have no side effects.
+The *countable* properties do not include *explicit implementations* of `ICollection<T>.Count` or `ICollection.Count`.
+
+If the target type is a *struct* or *class type* that implements `System.Collections.IEnumerable`, and the target type does not have a *[create method](#create-methods)*, the construction of the collection instance as follows:
+
+* If the collection expression does not contain *spread elements*, and the collection type contains an applicable constructor with a single `int capacity` parameter, the compiler *may* invoke that constructor with a non-zero capacity value. Otherwise, the constructor that is applicable with no arguments is invoked.
+
+* For each element in order:
+  * If the element is an *expression element*, the applicable `Add` instance or extension method is invoked with the element *expression* as the argument. (Interleaving `Add` calls matches classic [*collection initializer behavior*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#117154-collection-initializers).)
+  * If the element is a *spread element* then:
+    * If the target type has an `AddRange` instance or extension method that is applicable when invoked with the spread element *expression*, the `AddRange` method *may be* invoked with expression as the argument.
+    * Otherwise, the spread element *expression* is iterated using the applicable `GetEnumerator` instance or extension method, and the applicable `Add` instance or extension method is invoked for each item from the enumerator in order.
+
+If the target type is an *array*, a *span*, a type with a *[create method](#create-methods)*, or an *interface*, the construction of the collection instance is as follows:
```

Comment:
Is the intent that any other `ref struct` type other than `Span<T>` and `ReadOnlySpan<T>` use a *create_method*? Or should the language be modified so *span* means any `ref struct` type that matches the proper shape?

Reply:
A `ref struct` type could be implemented using the *collection initializer type* pattern or it could have a *create method*. The intent here was to be explicit about the expected types other than *collection initializer types*.